src/systems/filecoin_markets/storage_market/storage_market.id

```diff
@@ -0,0 +1,79 @@
+import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
+import block "github.com/filecoin-project/specs/systems/filecoin_blockchain/struct/block"
+import actor "github.com/filecoin-project/specs/systems/filecoin_vm/actor"
+import piece "github.com/filecoin-project/specs/systems/filecoin_files/piece"
+import deal "github.com/filecoin-project/specs/systems/filecoin_markets/storage_market/storage_deal"
+import ipld "github.com/filecoin-project/specs/libraries/ipld"
+
+type StorageDealProposalNotFound struct {}
+type StorageDealProposalRejected struct {}
+type StorageDealProposalAccepted struct {}
+type StorageDealProposalSigned struct {}
+type StorageDealPublished struct {}
+type StorageDealCommitted struct {}
+type StorageDealActive struct {}
+type StorageDealFailing struct {}
+type StorageDealRecovering struct {}
+type StorageDealExpired struct {}
+type StorageDealNotFound struct {}
+
+type StorageDealStatus union {
+    StorageDealProposalNotFound
+    StorageDealProposalRejected
+    StorageDealProposalAccepted
+    StorageDealProposalSigned
+    StorageDealPublished
+    StorageDealCommitted
+    StorageDealActive
+    StorageDealFailing
+    StorageDealRecovering
+    StorageDealExpired
+    StorageDealNotFound
+}
+
+type Signature struct {}
+
+type StorageAsk struct {
+    Price         actor.TokenAmount  // attoFIL per GiB per epoch
+    Collateral    actor.TokenAmount  // attoFIL per GiB per epoch
+
+    MinPieceSize  piece.PieceSize
+    Miner         addr.Address
+    Timestamp     block.ChainEpoch
+    MaxDuration   block.ChainEpoch
+    MinDuration   block.ChainEpoch
+    SeqNo         UVarint
+}
+
+type SignedStorageAsk struct {
+    Ask StorageAsk
+    Signature
+}
+
+type AskRequest struct {
+    Miner addr.Address
+}
+
+type AskResponse struct {
+    Ask SignedStorageAsk
+}
+
+type DealRequest struct {
```

Comment:
maybe we can find a better name than `DealRequest` to understand the purpose of this object...

Why not just include the payload root in the `DealProposal`?

Reply:
DealProposal is a StorageDealProposal, which will ultimately go on-chain for the deal. For a variety of reasons, my understanding is the Payload is not part of what goes on chain (it doesn't have anything to do with future PoST/PoRep)

Also, I don't quite know the state of this, but there has been talk at various points that we would support deals based on types of data other than IPLD.

Reply:
ok ignore me then. that sgtm